src/IteratorImpl.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
```

Comment:
Should be IteratorImpl.hpp . But do we really have to include it from somewhere? You may put this code inside .cpp